pkg/cluster/internal/providers/docker/provider.go

```diff
@@ -276,6 +276,28 @@ func (p *provider) Info() (*providers.ProviderInfo, error) {
 	return p.info, err
 }
 
+// ContainerSave saves images to dest, as in `docker save`
+func (p *provider) ContainerSave(images []string, dest string) error {
```

Comment:
Saving images from storage isn't really a property of implementing nodes, and this is already a pretty complicated object. I think we should probably split this out.